File: plugin-rust-agent/src/main/java/jetbrains/buildServer/rust/logging/CargoLoggerFactory.java
/*
 * Copyright 2000-2016 JetBrains s.r.o.
 *
 * Licensed under the Apache License, Version 2.0 (the "License").
 * See LICENSE in the project root for license information.
 */

package jetbrains.buildServer.rust.logging;

import jetbrains.buildServer.agent.BuildProgressLogger;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

import java.util.HashMap;
import java.util.Map;

/**
 * @author Dmitry.Tretyakov
 *         Date: 25.09.2016
 *         Time: 21:49
 */
public class CargoLoggerFactory {
    private final Map<CargoState, CargoLogger> myLoggers;

    public CargoLoggerFactory(@NotNull final BuildProgressLogger logger){
        myLoggers = new HashMap<CargoState, CargoLogger>();
        myLoggers.put(CargoState.Running, new CargoStateLogger(logger, CargoState.Running));
        myLoggers.put(CargoState.Compiling, new CargoCompileLogger(logger));
        myLoggers.put(CargoState.Error, new CargoStateLogger(logger, CargoState.Error));
        myLoggers.put(CargoState.Warning, new CargoStateLogger(logger, CargoState.Warning));
        myLoggers.put(CargoState.Documenting, new CargoStateLogger(logger, CargoState.Documenting));
        myLoggers.put(CargoState.Fresh, new CargoStateLogger(logger, CargoState.Fresh));
        myLoggers.put(CargoState.Updating, new CargoStateLogger(logger, CargoState.Updating));
        myLoggers.put(CargoState.Adding, new CargoStateLogger(logger, CargoState.Adding));
        myLoggers.put(CargoState.Removing, new CargoStateLogger(logger, CargoState.Removing));
        myLoggers.put(CargoState.DocTests, new CargoStateLogger(logger, CargoState.DocTests));
        myLoggers.put(CargoState.Packaging, new CargoStateLogger(logger, CargoState.Packaging));
        myLoggers.put(CargoState.Downloading, new CargoStateLogger(logger, CargoState.Downloading));
        myLoggers.put(CargoState.Uploading, new CargoStateLogger(logger, CargoState.Uploading));
        myLoggers.put(CargoState.Verifying, new CargoStateLogger(logger, CargoState.Verifying));
        myLoggers.put(CargoState.Archiving, new CargoStateLogger(logger, CargoState.Archiving));
        myLoggers.put(CargoState.Installing, new CargoStateLogger(logger, CargoState.Installing));
        myLoggers.put(CargoState.Replacing, new CargoStateLogger(logger, CargoState.Replacing));
        myLoggers.put(CargoState.Default, new CargoDefaultLogger(logger));
        myLoggers.put(CargoState.Testing, new CargoTestingLogger(logger));
    }

    @NotNull
    public CargoLogger getLogger(final CargoState state) {
        return myLoggers.get(state);
    }
}

File: plugin-rust-agent/src/main/kotlin/jetbrains/buildServer/rust/logging/CargoLoggerFactory.kt


package jetbrains.buildServer.rust.logging

import jetbrains.buildServer.agent.BuildProgressLogger

/**
 * Logger factory.
 */
class CargoLoggerFactory(val logger: BuildProgressLogger) {
    private val myLoggers = mapOf(
            Pair(CargoState.Running, CargoStateLogger(logger, CargoState.Running)),
            Pair(CargoState.Compiling, CargoCompileLogger(logger)),
            Pair(CargoState.Created, CargoStateLogger(logger, CargoState.Created)),
            Pair(CargoState.Finished, CargoStateLogger(logger, CargoState.Finished)),
            Pair(CargoState.Error, CargoStateLogger(logger, CargoState.Error, "ERROR")),
            Pair(CargoState.Warning, CargoStateLogger(logger, CargoState.Warning, "WARNING")),
            Pair(CargoState.Documenting, CargoStateLogger(logger, CargoState.Documenting)),
            Pair(CargoState.Fresh, CargoStateLogger(logger, CargoState.Fresh)),
            Pair(CargoState.Updating, CargoStateLogger(logger, CargoState.Updating)),
            Pair(CargoState.Adding, CargoStateLogger(logger, CargoState.Adding)),
            Pair(CargoState.Removing, CargoStateLogger(logger, CargoState.Removing)),
            Pair(CargoState.DocTests, CargoStateLogger(logger, CargoState.DocTests)),
            Pair(CargoState.Packaging, CargoStateLogger(logger, CargoState.Packaging)),
            Pair(CargoState.Downloading, CargoStateLogger(logger, CargoState.Downloading)),
            Pair(CargoState.Uploading, CargoStateLogger(logger, CargoState.Uploading)),
            Pair(CargoState.Verifying, CargoStateLogger(logger, CargoState.Verifying)),
            Pair(CargoState.Archiving, CargoStateLogger(logger, CargoState.Archiving)),
            Pair(CargoState.Installing, CargoStateLogger(logger, CargoState.Installing)),
            Pair(CargoState.Replacing, CargoStateLogger(logger, CargoState.Replacing)),
            Pair(CargoState.Unpacking, CargoStateLogger(logger, CargoState.Unpacking)),
            Pair(CargoState.Testing, CargoTestingLogger(logger)),
            Pair(CargoState.ErrorDetails, CargoErrorDetailsLogger(logger)))

    fun getLogger(state: CargoState): CargoLogger {
        return myLoggers[state] ?: CargoDefaultLogger(logger)
    }
}